lib/ansible/plugins/inventory/__init__.py

```diff
@@ -292,21 +293,28 @@ def _add_host_to_keyed_groups(self, keys, variables, host, strict=False):
             for keyed in keys:
                 if keyed and isinstance(keyed, dict):
                     prefix = keyed.get('prefix', '')
+                    if prefix:
+                        prefix = prefix + "_"
                     key = keyed.get('key')
                     if key is not None:
                         try:
-                            groups = to_safe_group_name('%s_%s' % (prefix, self._compose(key, variables)))
+                            groups = self._compose(key, variables)
```

Comment:
removing this will create a bunch of uncaught exceptions down the line

Reply:
I can wrap the call to `to_safe_group_name` in the same exception handling.

Reply:
What exceptions are generated by `to_safe_group_name` - it just calls `re.sub` doesn't it? (assuming I don't cause the errors @s-hertel mentions through lack of ensuring that groups is a list rather than a stringtype)

Reply:
the exceptions will happen in the plays when people try to access invalid group names

Reply:
There shouldn't be any unsafe group names - the conversion to safe group names just happens slightly later.